### PR TITLE
docs: initial content for references.d.ts

### PR DESCRIPTION
**Pull request by Ombuweb**

### CODE REVIEW

**Comment by rigor789** [comment]
We need to document more about `@nativescript/types` however I'm unsure whether or not it should be on the references.d.ts page here, or a separate page we link to from this page?

Curious about everyone's take on that /cc @NativeScript/tsc

**Comment by NathanWalker** [comment]
How about we include under the package.json since it is a dependency?

- references.d.ts 
- package.json
  - @nativescript/core
  - @nativescript/types
  - @nativescript/webpack

Within project structure. The references.d.ts can be shown in the project structure but would jump to the @nativescript/types page to explain what they are and how/why references.d.ts plays into it.

**Comment by NathanWalker** [comment]
Reason for this is I recently brought up on TSC about moving away from global declares from types which would mean we could possibly move away from references.d.ts in the future - that way having it link to @nativescript/types page would work for now and future.

File: content/project-structure/references-d-ts.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+title: references.d.ts
+---
+
+The `references.d.ts` file contains the types definitions for the APIs accessed by the NativeScript app.
```
Comment (NathanWalker) [inline content/project-structure/references-d-ts.md]:
```suggestion
The `references.d.ts` file contains the TypeScript declarations for the platform APIs accessed by the NativeScript app.
```

**Pull request merged**